demos/bio/client-arg.c

```diff
@@ -46,13 +46,32 @@ int main(int argc, char **argv)
             continue;
         /* Otherwise application specific argument processing */
         if (strcmp(*args, "-connect") == 0) {
+            if (nargs < 2) {
+                fprintf(stderr, "Missing argument after -connect\n");
+                goto end;
+            }
             connect_str = args[1];
-            if (connect_str == NULL) {
-                fprintf(stderr, "Missing -connect argument\n");
+            int colon_pos = strcspn(connect_str, ":");
```

Comment:
Declaration need to go before code.  We target C89 which doesn't permit intermixed declarations and code.